test/com/esotericsoftware/kryo/serializers/CompatibleFieldSerializerTest.java

```diff
@@ -642,4 +653,117 @@ public int hashCode () {
 			return Objects.hash(value, list, serializable);
 		}
 	}
+	// #699 check ExtendedFieldName for CompatibleFieldSerializer
+	static public class ExtendedFieldNamesTest {
+		@Test(expected = IllegalArgumentException.class)
+		public void setExtendedFieldNamesDefault() throws IOException {
+			final Child child = new Child();
+			child.setCustomNote(true);
+			child.setCid(3);
+			Kryo kryo = getKryo(false);
+			final File outputFile = File.createTempFile("temp_output", "dat");
+			try (final Output output = new Output(new FileOutputStream(outputFile))) {
+				kryo.writeObject(output, child);
+			}
+		}
+
+		@Test
+		public void setExtendedFieldNamesIsTrue() throws IOException {
+			final Child child = new Child();
+			child.setCustomNote(true);
+			child.setCid(3);
+			Kryo kryo = getKryo(true);
+			final File outputFile = File.createTempFile("temp_output", "dat");
+			try (final Output output = new Output(new FileOutputStream(outputFile))) {
+				kryo.writeObject(output, child);
+				final Input input = new Input(new FileInputStream(outputFile));
+				final Child restoreChild = kryo.readObject(input, Child.class);
+				Assert.assertEquals(true, restoreChild.customNote);
+				input.close();
+			}
+		}
+
+		private Kryo getKryo(Boolean isSetExtendedFieldNames) {
+			final Kryo kryo = new Kryo();
+			kryo.setDefaultSerializer(new SerializerFactory.BaseSerializerFactory() {
+				@Override
+				public Serializer newSerializer(Kryo kryo, Class type) {
+					final CompatibleFieldSerializer.CompatibleFieldSerializerConfig config =
+							new CompatibleFieldSerializer.CompatibleFieldSerializerConfig();
+					config.setChunkedEncoding(true);
+					config.setReadUnknownFieldData(true);
+					if (isSetExtendedFieldNames)
+						config.setExtendedFieldNames(true);
+					final CompatibleFieldSerializer serializer = new CompatibleFieldSerializer(kryo, type, config);
+					return serializer;
+				}
+			});
+			kryo.setRegistrationRequired(false);
+			kryo.setInstantiatorStrategy(new DefaultInstantiatorStrategy(new SerializingInstantiatorStrategy()));
+			return kryo;
+		}
+
+		private class Child extends Father {
```

Comment:
Please make the classes `static` and use more generic class and field names. Something like `ClassWithDuplicateField`, `SuperClass`, `SuperSuperClass`.